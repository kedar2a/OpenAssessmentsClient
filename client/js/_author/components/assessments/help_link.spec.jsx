import React        from 'react';
import { shallow }  from 'enzyme';
import HelpLink   from './help_link';

describe('help_link component', () => {
  let props;
  let result;

  beforeEach(() => {
    props = {
      to: '/somewhere.html',
    };
    result = shallow(<HelpLink {...props} />);
  });

  it('matches the snapshot taken', () => {
    expect(result).toMatchSnapshot();
  });

  it('should also render with an icon', () => {
    expect(result.text()).toContain('open_in_new');
    expect(result.find('.material-icons').length).toBe(1);
    props = {
      to: '/somewhere.html',
      icon: true
    };
    result = shallow(<HelpLink {...props} />);
    expect(result.text()).toContain('help');
    expect(result.find('.material-icons').length).toBe(1);
  });
});
