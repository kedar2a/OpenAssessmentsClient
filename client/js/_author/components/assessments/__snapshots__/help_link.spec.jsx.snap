// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`help_link component matches the snapshot taken 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <a
    className="help-link"
    href="/somewhere.html"
    rel="noreferrer noopener"
    target="_blank"
>
    What's this?
    <i
        aria-hidden={true}
        className="material-icons open-external"
    >
        open_in_new
    </i>
</a>,
  "nodes": Array [
    <a
      className="help-link"
      href="/somewhere.html"
      rel="noreferrer noopener"
      target="_blank"
>
      What's this?
      <i
            aria-hidden={true}
            className="material-icons open-external"
      >
            open_in_new
      </i>
</a>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <helpLink
        to="/somewhere.html"
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "to": "/somewhere.html",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <a
          className="help-link"
          href="/somewhere.html"
          rel="noreferrer noopener"
          target="_blank"
>
          What's this?
          <i
                    aria-hidden={true}
                    className="material-icons open-external"
          >
                    open_in_new
          </i>
</a>,
        "_debugID": 2,
        "_renderedOutput": <a
          className="help-link"
          href="/somewhere.html"
          rel="noreferrer noopener"
          target="_blank"
>
          What's this?
          <i
                    aria-hidden={true}
                    className="material-icons open-external"
          >
                    open_in_new
          </i>
</a>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <helpLink
    to="/somewhere.html"
/>,
}
`;
